UI/EncyclopediaDetailPanel.cpp

```diff
@@ -949,9 +985,11 @@ void EncyclopediaDetailPanel::HandleSearchTextEntered() {
     std::set<std::pair<std::string, std::string>> already_listed_results;  // pair of category / article name
 
     // assemble link text to all pedia entries, indexed by name
-    std::vector<std::string> dir_names = GetSearchTextDirNames();
-    for (const std::map<std::string, std::vector<EncyclopediaArticle>>::value_type& article : GetEncyclopedia().articles) {
-        dir_names.push_back(article.first);
+    std::unordered_set<std::string> dir_names;
```

Comment:
Not seeing where this occurs? `dir_names` is not iterated over until `all_pedia_entries_list` is built.
I've renamed `dir_name` to `topdir_name` in case it caused confusion.

Circular categories are possible, though they should not display to begin with.